### PR TITLE
Closes #146: Interactive exploration of datasets vignette in website

### LINKED ISSUE #146: Documentation: Allow interactive exploration of datasets in referrence pages/vignett
### Please select a category the issue is focused on?

_No response_

### Let us know where something needs a refresh or put your idea here!

It would be nice to be able to peruse each dataset within a vignette or the reference page. Open to ideas on how to achieve this!

Note: this issue is similar to the [corresponding one in pharmaverseadam](https://github.com/pharmaverse/pharmaverseadam/issues/102).

### PR DESCRIPTION
**Pull request by Gero1999**
> Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   
Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

## Implementation description
This PR:
- Added a new vignette, `preview-sdtm.Rmd`, which provides an interactive, filterable preview of all datasets in the `pharmaversesdtm` package, displaying the first 50 rows of each dataset in collapsible panels using DT package.
-  Updated the site navigation in `_pkgdown.yml` to add a "Preview SDTM" link in the navbar, pointing to the new vignette.
- Included the vignettes folder so they are ignored from the CRAN build

___

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build pharmaversesdtm site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/cran-release/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Review by jimrothstein — APPROVED** [review]
Impressive!    Ran fine in Rstudio cloud.    I did not yet go through your code.

- Change "Show XX entries"   100 is maximum in the drop-down.  Change to 50 ? because  actual limit is 50
- On my screen, there is large left margin.   Possible to remove?
- Introductory text is fine.    
- Just nitpicking here:  
-  "preview of the datasets"  change to "preview of **all**  datasets"
-  Text is all black and white;   no indication user should click.
- Several datasets are less than 50.  Maybe I could add the number of rows, like
```
- ce_vaccine (44):  Clinical Events for Vaccine
```

Will go through your code now.

Image

**Comment by Gero1999** [comment]
> image

I might have missed to load the package. I will give it a quick look and see for an easy fix

**Comment by Gero1999** [comment]
hopefully now is fixed! otherwise I will give it a deeper look later

**Comment by manciniedoardo** [comment]
@pharmaverse/admiral take a look at this PR - looks like a super useful addition to our data package. @Fanny-Gautier definitely something to mimic in pharmaverseadam once merged!

**Comment by manciniedoardo** [comment]
> @Gero1999, One question: can we create a single table that allows filtering by TA and shows the respective datasets with previews, instead of listing the datasets? Right now, it looks similar to the reference page.

I agree it looks similar to the reference page, but i think some overlap is ok if the implementation is as simple as this. in my review i left some comments to help differentiate - but final call up to you @Lina2689

**Comment by Gero1999** [comment]
> @pharmaverse/admiral take a look at this PR - looks like a super useful addition to our data package. @Fanny-Gautier definitely something to mimic in pharmaverseadam once merged!

if you want @manciniedoardo @Fanny-Gautier I can also help you translating it into `pharmaverseadam`, I gave it a look and should be straightforward

Regarding the comments I will try to address them later today or tomorrow asap. Glad to hear that you like it ^^

**Review by bms63 — COMMENTED** [review]
Is this being made as an article rather than as a vignette?  I would be worry about this passing the CRAN filters.

File: vignettes/articles/preview-sdtm.Rmd [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+Here is a preview of all datasets included in the `{pharmaversesdtm}` package. Organized by therapeutic area, each dataset is displayed in a collapsible panel with filtering options for easier exploration. For simplicity, only the first 50 rows of each dataset are shown.
+
+```{r echo=FALSE}
+library(pharmaversesdtm)
+library(htmltools)
+library(reactable)
+library(yaml)
+
+specs_yml <- yaml::read_yaml(system.file("extdata/sdtms-specs.json", package = "pharmaversesdtm"))
```
Comment (Fanny-Gautier) [inline vignettes/articles/preview-sdtm.Rmd]: Is `yaml::` call needed with `library(yaml)` call at line 19?

**Comment by Fanny-Gautier** [comment]
> Also - way cool!!
> 
> Just musing - would it be easier to have as a standalone website for both sdtm and adam that is linked to this pkgdown site? That way the maintenance/views could be experimented with updated and not depend on a release??
> 
> Unsure if this would get release when available or have to wait for a pkg release...feels like a great idea to get out the door quickly :)

The plan was to issue an ad‑hoc release in the spring to include the microbiology data, the new vignette will therefore be available at that time as well.

**Comment by bms63** [comment]
> > Also - way cool!!
> > Just musing - would it be easier to have as a standalone website for both sdtm and adam that is linked to this pkgdown site? That way the maintenance/views could be experimented with updated and not depend on a release??
> > Unsure if this would get release when available or have to wait for a pkg release...feels like a great idea to get out the door quickly :)
> 
> The plan was to issue an ad‑hoc release in the spring to include the microbiology data, the new vignette will therefore be available at that time as well.

if the package isn't changing, you could delete the release and do a "new release" with the vignette and that will trigger the build of website with the vignette.  be so cool to have it out before spring time!  it is such a neat addition.

**Review by Lina2689 — APPROVED** [review]
Looks good!

**Comment by Lina2689** [comment]
@manciniedoardo Could you please approve the PR? I’ll merge it right after approval.

**Review by manciniedoardo — APPROVED** [review]
